### PR TITLE
Fixes #215 - Automation of monthly AIRAC tasks

### LINKED ISSUE #215: AIRAC related tasks automation
**Is your feature request related to a problem? Please describe.**
Yes, boring automatable tasks that happen every 4 weeks.

**Describe the solution you'd like**
I'd like to do one of two things, with the latter being the preferred option (but likely more complicated)

Easy:
Raise an issue every wednesday before AIRAC day (once every 4 weeks) that details what is required to do to include the Sector File.
1. Replace SF with latest version
2. Replace all references in .prf files to point to new one
3. Replace ICAO_ datafiles

Hard:
Get the sector file from the Sector file repo when a release happens, and automate some of the tasks above. Unable to do the ICAO datafiles as it's pulled from GNG.

**Is your suggestion based on a real world display?**
N/A

**Additional context**

### PR DESCRIPTION
**Pull request by chssn**
Fixes #215 

# Summary of changes

- [x] Replace SF with latest version
- [x] Replace all references in .prf files to point to new one
- [x] Replace ICAO_ datafiles ~~- waiting to arrange meeting with @luke11brown although this shouldn't delay the first release.~~

**auto_airac_actions.yml**
~~1. Wait for webhook from VATSIM-UK/UK-Sector-File OR on push to main OR manual trigger~~
1. Wait for "Compile and Validate" on [UK-Sector-File](https://github.com/VATSIM-UK/UK-Sector-File) to trigger
2. Checkout VATSIM-UK/uk-controller-pack:main
3. Run **auto-airac-actions.py**
4. Commit changes as a new branch in the format **sf_[\d]{4}/[\d]{2}** eg sf_2023/06
5. Create a pull request (see example in screenshot below)
6. Auto-assign reviewer(s) - _will need a list of reviewers here - set to chssn for testing at the moment_

This relies on a ~~webhook being generated on~~ successful compile of [UK-Sector-File](https://github.com/VATSIM-UK/UK-Sector-File). This also requires both repos having personal access tokens (PAT) generated and stored as a secret in the opposing repo. Both PATs need to have [read/write repo scope](https://github.com/marketplace/actions/repository-dispatch).

1. The PAT for UK-Sector-File repo needs to be stored as `DOWNLOAD_ARTIFACT` secret in the uk-controller-pack repo
2. The PAT for uk-controller-pack repo needs to be stored as `UK_CONTROLLER_PACK_ACCESS_TOKEN` secret in the UK-Sector-File repo.

**auto_airac_actions.py**
1. Calculate the current AIRAC release
3. Compare this with the sector file found after checkout VATSIM-UK/uk-controller-pack:main
~~3. If different, attempt to download latest sector file from http://www.vatsim.uk/files/sector/esad/~~
4. Download the latest sector file artefact from [UK-Sector-File](https://github.com/VATSIM-UK/UK-Sector-File)
5. Unzip and copy files with appropriate name changes (eg UK.ese to UK_2023_06.ese)
6. Iterate over all *.asr and *.prf files making changes to or verifying the sector file references
7. Clean up any old sector file data

# Screenshots (if necessary)

image

### CODE REVIEW

**Comment by luke11brown** [comment]
Excellent work @chssn Would it be easier to base the trigger on when a release occurs in the Sector File GitHub repo, rather than looking at the UK Website?

Unsure of some of the technicalities, but the release creates artifacts on GitHub which can be extracted as part of your script. The file hosted on the VATUK website relies on myself/Peter uploading manually.

I can run you through the GNG process on Discord, as it's a bit convoluted.

**Comment by luke11brown** [comment]
ASR files shouldn't contain any references to the sector file or its' path. Only reference should be in .prf files, as is the case in the current pack.

From what I can see in the test PR, the script is also adding in the lines in ASR files.

**Comment by luke11brown** [comment]
Agreed that the above works correctly, however, I believe there to be the following point of failure:

In the case the pack isn't updated in line with the sector file, the ASRs will reference, let's say for example 2023_05.
If the user has auto-updated the sector file to the latest version, e.g. 2023_06 or later...
Profiles/ASRs that are yet to be opened will look for the old SECTORTITLE in the ASR, fail to do so and will start blank.

Unfortunately, IMHO the best way is to keep the references in only the PRFs. Whilst we encourage users to use the folder specified in the readme, it should be somewhat portable (within the limitations of ES).

We still have lots of users with older versions of ES running due to issues running Sweatbox scenarios and I'm keen not to break compatibility with those by making the pack only run from %appdata%.

Whilst I agree, that Gergely is following best practice by following MS's advice, the way that ES requires selection and manual placement of files in %appdata% doesn't lend itself to users who aren't familiar with Windows or may not have access to unhide folders.

File: .github/workflows/auto_airac_actions.py [hunk 1]
```diff
@@ -91,6 +92,59 @@ def __init__(self) -> None:
...
+        list_of_files = [
+            "ICAO/ICAO_Aircraft.txt",
+            "ICAO/ICAO_Airlines.txt",
+            "ICAO/ICAO_Airports.txt",
+            "NavData/airway.txt",
+            "NavData/icao.txt",
+            "NavData/isec.txt"
+        ]
+        for file in list_of_files:
+            shutil.move(f"import/EGTT/{file}", f"UK/Data/Datafiles/{file.split('/', maxsplit=1)[-1]}")
```
Comment (luke11brown) [inline .github/workflows/auto_airac_actions.py:142]: THere is also an ICAO_AIrlines in the vSMR folder

**Comment by luke11brown** [comment]
@AndyTWF Can't add you as a reviewer, but I defer to you as the higher power.

**Comment by CalumTowers** [comment]
Python and the context of this isn’t my area of expertise so I’ll defer to others who may be closer to support!

**Comment by luke11brown** [comment]
> Python and the context of this isn’t my area of expertise so I’ll defer to others who may be closer to support!

I will however need perms to create the tokens in both repos as described, unless you can create them?

**Comment by luke11brown** [comment]
> Python and the context of this isn’t my area of expertise so I’ll defer to others who may be closer to support!

I will however need perms to create the tokens in both repos as described, unless you can create them?

**Comment by luke11brown** [comment]
> Python and the context of this isn’t my area of expertise so I’ll defer to others who may be closer to support!

@CalumTowers I will however need perms to create the tokens in both repos as described, unless you can create them?

File: .github/workflows/auto_airac_actions.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+
+      - name: Install dependencies
+        run: pip install -r .github/workflows/requirements.txt
+
+      - name: Run Python script
+        env:
+            REMOTE_KEY: ${{ secrets.DOWNLOAD_ARTIFACT }}
```
Comment (VATSIMUK) [inline .github/workflows/auto_airac_actions.yml:28]:
```suggestion
            REMOTE_KEY: ${{ secrets.GITHUB_TOKEN }}
```
Using the `GITHUB_TOKEN` should authenticate the API request that this is being used for without the need for a specific PAT.

https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow

**Review by luke11brown — APPROVED** [review]

**Pull request merged**